src/timescale/mod.rs

```diff
@@ -217,61 +192,45 @@ impl From<u8> for TimeScale {
     }
 }
 
-impl FromStr for TimeScale {
-    type Err = Errors;
+#[cfg(test)]
+mod unit_test_timescale {
+    use super::TimeScale;
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_serdes() {
+        let ts = TimeScale::UTC;
+        let content = "\"UTC\"";
+        assert_eq!(content, serde_json::to_string(&ts).unwrap());
+        let parsed: TimeScale = serde_json::from_str(content).unwrap();
+        assert_eq!(ts, parsed);
+    }
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let val = s.trim();
-        if val == "UTC" {
-            Ok(Self::UTC)
-        } else if val == "TT" {
-            Ok(Self::TT)
-        } else if val == "TAI" {
-            Ok(Self::TAI)
-        } else if val == "TDB" {
-            Ok(Self::TDB)
-        } else if val == "ET" {
-            Ok(Self::ET)
-        } else if val == "GPST" || val == "GPS" {
-            Ok(Self::GPST)
-        } else if val == "GST" || val == "GAL" {
-            Ok(Self::GST)
-        } else if val == "BDT" || val == "BDS" {
-            Ok(Self::BDT)
-        } else if val == "QZSST" || val == "QZSS" {
-            Ok(Self::QZSST)
-        } else {
-            Err(Errors::ParseError(ParsingErrors::TimeSystem))
+    #[test]
+    fn test_ts() {
+        for ts_u8 in 0..u8::MAX {
+            let ts = TimeScale::from(ts_u8);
+            let ts_u8_back: u8 = ts.into();
+            // If the u8 is greater than 5, it isn't valid and necessarily encoded as TAI.
+            if ts_u8 < 9 {
+                assert_eq!(ts_u8_back, ts_u8, "got {ts_u8_back} want {ts_u8}");
+            } else {
+                assert_eq!(ts, TimeScale::TAI);
+            }
         }
     }
-}
 
-#[test]
-#[cfg(feature = "serde")]
-fn test_serdes() {
-    let ts = TimeScale::UTC;
-    let content = "\"UTC\"";
-    assert_eq!(content, serde_json::to_string(&ts).unwrap());
-    let parsed: TimeScale = serde_json::from_str(content).unwrap();
-    assert_eq!(ts, parsed);
-}
-
-#[test]
-fn test_ts() {
-    for ts_u8 in 0..u8::MAX {
-        let ts = TimeScale::from(ts_u8);
-        let ts_u8_back: u8 = ts.into();
-        // If the u8 is greater than 5, it isn't valid and necessarily encoded as TAI.
-        if ts_u8 < 9 {
-            assert_eq!(ts_u8_back, ts_u8, "got {ts_u8_back} want {ts_u8}");
-        } else {
-            assert_eq!(ts, TimeScale::TAI);
+    #[test]
+    fn test_initial_year() {
```

Comment:
I refactored `ref_year()`, renamed to `initial_year()` which is intended to be used in Compute_from_gregorian, mostly when formatting.